File: Lesson5 Sorting/Problem4/Solutions/Solution.java
import java.util.Scanner;
public class Solution {
	public static void main(String[]args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt(), array[] = new int[n]; //n is the length of the array
		for(int i = 0; i < n; i++) {
			array[i] = sc.nextInt();
		}
		
		for(int j = 0; j < array.length; j++) {
			System.out.print(array[j] + " ");
		}
		System.out.println(); //print on a new line
		
		boolean isSorted = false; //set a boolean to check if the array is sorted
		while(!isSorted) { //while the array is not sorted,
			isSorted = true;
			for(int i = 0; i < n - 1; i++) {
				if(array[i] > array[i + 1]) { //bubble sort
					int temp = array[i];
					array[i] = array[i + 1];
					array[i + 1] = temp;
					isSorted = false;
					for(int j = 0; j < n; j++) {
						System.out.print(array[j] + " ");
					}
					System.out.println(); //print on a new line
				}
			}
		}
		sc.close();
	}
}

File: Lesson5 Sorting/Problem4/Solutions/Solution.cpp
#include <iostream>

using namespace std;

int main()
{
    int l; cin>>l; int array[l];
    for(int i = 0; i < l; i++) {
        cin>>array[i];
    }
    for(int i = 0; i < l; i++){
        cout<<array[i]<<" ";
    }
    cout<<endl;
    bool isSorted = false;
    while(!isSorted){
        isSorted = true;
        for(int i = 0; i < l - 1; i++){
            if(array[i] > array[i + 1]){
                int temp = array[i];
                array[i] = array[i + 1];
                array[i + 1] = temp;
                isSorted = false;
                for(int j = 0; j < l; j++){
                    cout<<array[j]<<" ";
                }
                cout<<endl;
            }
        }
    }
    return 0;
}
